source/plugins/go/src/extension/extension.go

```diff
@@ -320,6 +320,91 @@ func (e *Extension) GetNamespaceFilteringModeForDataCollection() string {
 	return namespaceFilteringMode
 }
 
+func (e *Extension) GetContainerLogV2ExtensionConfig(isWindows bool) (map[string][]string, map[string]string, error) {
+	namespaceStreamIdsMap := make(map[string][]string)
+	StreamIdNamedPipeMap := make(map[string]string)
+
+	guid := uuid.New()
+	var extensionData TaggedData
+	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": "ContainerLogV2Extension", "Version": "1"}
+	jsonBytes, err := json.Marshal(taggedData)
+	if err != nil {
+		logger.Printf("extension::GetContainerLogV2ExtensionConfig::Failed to marshal taggedData data. Error message: %s", string(err.Error()))
+		return namespaceStreamIdsMap, StreamIdNamedPipeMap, err
+	}
+
+	responseBytes, err := getExtensionConfigResponse(jsonBytes)
+	if err != nil {
+		logger.Printf("extension::GetContainerLogV2ExtensionConfig::Failed to get config response data. Error message: %s", string(err.Error()))
+		return namespaceStreamIdsMap, StreamIdNamedPipeMap, err
+	}
+	var responseObject AgentTaggedDataResponse
+	err = json.Unmarshal(responseBytes, &responseObject)
+	if err != nil {
+		logger.Printf("extension::GetContainerLogV2ExtensionConfig::Failed to unmarshal config response data. Error message: %s", string(err.Error()))
+		return namespaceStreamIdsMap, StreamIdNamedPipeMap, err
+	}
+
+	err = json.Unmarshal([]byte(responseObject.TaggedData), &extensionData)
+	extensionConfigs := extensionData.ExtensionConfigs
+	outputStreamDefinitions := make(map[string]StreamDefinition)
+	if isWindows {
+		outputStreamDefinitions = extensionData.OutputStreamDefinitions
+	}
+
+	for _, extensionConfig := range extensionConfigs {
+		outputStreamId := ""
+		namedPipe := ""
+		outputStreams := extensionConfig.OutputStreams
+		for dataType, outputStreamID := range outputStreams {
+			if strings.Compare(strings.ToLower(dataType), "containerinsights_containerlogv2") == 0 {
+				outputStreamId = outputStreamID.(string)
+				if isWindows {
+					namedPipe = outputStreamDefinitions[outputStreamID.(string)].NamedPipe
+					StreamIdNamedPipeMap[outputStreamId] = namedPipe
+					logger.Printf("GetContainerLogV2ExtensionConfig:: outputStreamId: %s namedPipe: %s", outputStreamId, namedPipe)
+				} else {
+					logger.Printf("GetContainerLogV2ExtensionConfig:: outputStreamId: %s", outputStreamId)
+				}
+			}
+		}
+
+		extensionSettings := extensionConfig.ExtensionSettings
+		dataCollectionSettingsItr, ok := extensionSettings["dataCollectionSettings"]
+
+		dataCollectionSettings := make(map[string]interface{})

```

Comment:
extension.go has a getDataCollectionSettingsInterface which do the similar thing

Reply:
> extension.go has a getDataCollectionSettingsInterface method which do the similar thing

refactored.